server/src/main/java/io/druid/server/router/CoordinatorRuleManager.java

```diff
@@ -206,6 +207,11 @@ private String getRuleURL()
       return null;
     }
 
-    return String.format("http://%s%s", server.getHost(), config.get().getRulesEndpoint());
+    return new URL(
```

Comment:
we should use URI instead of URL here to be consistent with the other methods.
URL also does DNS lookups which can be problematic.